packages/yarn-plugin/src/index.test.ts

```diff
@@ -0,0 +1,304 @@
+            targetRoot,
+            'packages/yarn-plugin/bundles/@yarnpkg/plugin-backstage.js',
+          ),
+        ],
+        {
+          cwd: mockDir.path,
+        },
+      );
+    }, 30_000);
+
+    beforeEach(() => {
+      mockDir.addContent({
+        'backstage.json': JSON.stringify({ version: '0.9.0' }),
+        packages: {
+          a: {
+            'package.json': JSON.stringify({
+              name: 'a',
+              dependencies: {
+                '@backstage/config': '^0.1.2',
+              },
+            }),
+          },
+          b: {
+            'package.json': JSON.stringify({
+              name: 'b',
+              dependencies: {
+                '@backstage/cli-common': '^0.1.1',
+                '@backstage/config': '^0.1.2',
+              },
+            }),
+          },
+        },
+      });
+    });
+
+    it('does not change yarn.lock when no backstage:^ versions are present', async () => {
+      await runYarnInstall();
+
+      await expect(
+        fs.readFile(joinPath(mockDir.path, 'yarn.lock'), 'utf-8'),
+      ).resolves.toEqual(initialLockFileContent);
+    }, 30_000);
+
+    describe('with backstage:^ dependencies', () => {
+      beforeEach(async () => {
+        mockDir.addContent({
+          'packages/a/package.json': JSON.stringify({
+            name: 'a',
+            dependencies: {
+              '@backstage/config': 'backstage:^',
+            },
+          }),
+          'packages/b/package.json': JSON.stringify({
+            name: 'b',
+            dependencies: {
+              '@backstage/config': 'backstage:^',
+              '@backstage/cli-common': 'backstage:^',
+            },
+          }),
+        });
+
+        await runYarnInstall();
+      }, 30_000);
+
+      it('retains existing lockfile entries', async () => {
+        const lockFileContent = await fs.readFile(
+          joinPath(mockDir.path, 'yarn.lock'),
+          'utf-8',
+        );
+
+        expect(nonWorkspaceEntries(lockFileContent)).toEqual(
+          nonWorkspaceEntries(initialLockFileContent),
+        );
+      });
+
+      it('bumps packages when backstage.json changes', async () => {
+        mockDir.addContent({
+          'backstage.json': JSON.stringify({
+            version: '1.0.0',
+          }),
+        });
+
+        await runYarnInstall();
+
+        const lockFileContent = await fs.readFile(
+          joinPath(mockDir.path, 'yarn.lock'),
+          'utf-8',
+        );
+
+        const lockFile = yaml.parse(lockFileContent);
+
+        // Versions from old manifest no longer appear in lockfile
+        expect(lockFile['@backstage/cli-common@npm:^0.1.1']).toBeUndefined();
+        expect(lockFile['@backstage/config@npm:^0.1.2']).toBeUndefined();
+
+        // Versions from new manifest have been added to lockfile
+        expect(lockFile['@backstage/cli-common@npm:^0.1.8']).toBeDefined();
+        expect(lockFile['@backstage/config@npm:^1.0.0']).toBeDefined();
+      }, 30_000);
```

Comment:
Might be worth just putting a `jest.setTimeout(30_000);` at the top instead

Reply:
Works for me!

Reply:
Done in 15de7a09c9cdf